src/frequenz/channels/timer.py

```diff
@@ -1,14 +1,90 @@
 # License: MIT
 # Copyright © 2023 Frequenz Energy-as-a-Service GmbH
 
-"""A timer receiver that ticks every `interval`.
+"""A receiver that sends a message regularly.
 
-Note:
-    This module always use `int`s to represent time.  The time is always in
-    microseconds, so the timer resolution is 1 microsecond.
+# Quick Start
 
-    This is to avoid floating point errors when performing calculations with
-    time, which can lead to very hard to reproduce, and debug, issues.
+If you need to do something as periodically as possible (avoiding
+[drifts](#missed-ticks-and-drifting)), you can use use
+a [`periodic()`][frequenz.channels.timer.Timer.periodic] timer.
+
+Example: Periodic Timer
+    ```python
+    import asyncio
+    from datetime import datetime, timedelta
+
+    from frequenz.channels.timer import Timer
+
+
+    async def main() -> None:
+        async for drift in Timer.periodic(timedelta(seconds=1.0)):
+            print(f"The timer has triggered at {datetime.now()} with a drift of {drift}")
+
+
+    asyncio.run(main())
+    ```
+
+If, instead, you need a timeout, for example to abort waiting for other receivers after
+a certain amount of time, you can use
+a [`timeout()`][frequenz.channels.timer.Timer.timeout] timer.
+
+Example: Timeout
+    ```python
+    import asyncio
+    from datetime import timedelta
+
+    from frequenz.channels import Anycast, select, selected_from
+    from frequenz.channels.timer import Timer
+
+
+    async def main() -> None:
+        channel = Anycast[int](name="data-channel")
+        data_receiver = channel.new_receiver()
+
+        timer = Timer.timeout(timedelta(seconds=1.0))
+
+        async for selected in select(data_receiver, timer):
+            if selected_from(selected, data_receiver):
+                print(f"Received data: {selected.value}")
+                timer.reset()
+            elif selected_from(selected, timer):
+                drift = selected.value
+                print(f"No data received for {timer.interval + drift} seconds, giving up")
+                break
+
+
+    asyncio.run(main())
+    ```
+
+    This timer will *rearm* itself automatically after it was triggered, so it will trigger
+    again after the selected interval, no matter what the current drift was.
```

Comment:
So `timeout` is actually also `periodic`.  And calling it a `timeout` timer doesn't clarify this.

Because when there are missed ticks, even the thing we call `periodic`, can't behave periodically, it just sends a burst of missed ticks, which is totally not periodic.  Triggering missed ticks makes the data look periodic, but it doesn't make the timer periodic. I guess software timers just can't be truly periodic, especially in a co-operative multi-tasking context.

They are both equally periodic or non-periodic based on whether there are missed ticks.  Maybe it was a bad decision to give names to these, and we should have exposed just the missed tick policies instead.

Sorry about bringing this up all over again, just ranting.  Feel free to ignore or defer until next major version.

Reply:
I see you've clarified these points below, but the names themselves aren't helping, I think.

Reply:
* https://github.com/frequenz-floss/frequenz-channels-python/issues/253